### DISCUSSION TITLE
[ERR] Error in nf90_get_var: SoilMoist_tavg Stopping.         -57

### DISCUSSION THREAD

**Discussion by sanjaync** [NASA-LIS/LISF#1592] [Q&A]
Hello sir, i tried to run step 4 to create cdf but I keep getting this error [ERR] Error in nf90_get_var: SoilMoist_tavg Stopping.         -57
[ldt.config.noahmp36_cdf.txt](https://github.com/user-attachments/files/16753503/ldt.config.noahmp36_cdf.txt)
[ldtlog.0000.txt](https://github.com/user-attachments/files/16753505/ldtlog.0000.txt)
[MPFilltest.log](https://github.com/user-attachments/files/16753506/MPFilltest.log)

**Comment by dmocko** [comment]
HI @sanjaync 

For the record, you are not running Step 4 of the Public Testcase Walkthrough, but rather you running a similar step,
having changed the LSM, among many other different options in the ldt.config file.

This is the error that you are seeing:

```
 [INFO] reading LSM output ../OL_OUTPUT/SURFACEMODEL/201701/LIS_HIST_201701010600.d01.nc
 [ERR] Error in nf90_get_var: SoilMoist_tavg Stopping.         -57
 endrun is being called
```

It's possible that this LIS_HIST file is not readable from the directory with your running LDT executable.
Please ensure that it is (and you have read permissions) by doing this in the directory where you run LDT:
   ls -l ../OL_OUTPUT/SURFACEMODEL/201701/LIS_HIST_201701010600.d01.nc

If that's not the issue, you likely didn't output the SoilMoist value as a time-average in your MODEL OUTPUT TBL.
Please ensure that the SoilMoist line in your MODEL OUTPUT TBL in your Step 2 lis.config file looks like:
`SoilMoist:    1  m3/m3   -    1 0 0 4  86 1000    # Average layer soil moisture (kg/m2)`

Note the 2nd and 5th columns should be "1" for the Step 4 CDF generation to work properly.
